pkg/openstack/common.go

```diff
@@ -7,12 +7,32 @@ import (
 
 	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
 	"github.com/openstack-k8s-operators/lib-common/modules/common/util"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
 	apimeta "k8s.io/apimachinery/pkg/api/meta"
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// CheckDeleteDisabled - Delete the object which in turn will clean the sub resources
+func CheckDeleteDisabled(ctx context.Context, helper *helper.Helper, obj client.Object) (ctrl.Result, error) {
+	key := client.ObjectKeyFromObject(obj)
+	if err := helper.GetClient().Get(ctx, key, obj); err != nil {
+		if k8s_errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+	// Delete the object
+	if obj.GetDeletionTimestamp().IsZero() {
+		if err := helper.GetClient().Delete(ctx, obj); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+	return ctrl.Result{}, nil
```

Comment:
Here we know that the delete succeeded. I think we should remove the service specific condition from the OpenStackControlPlane.Status.Condition list not to leave stale status about this deleted service there.

Reply:
Yeah, that sounds reasonable. I did not think about it earlier.

Reply:
I have pushed a PR[1] to remove a condition in lib-common to use here.

[1] https://github.com/openstack-k8s-operators/lib-common/pull/156